### PR TITLE
[USDU-269] Fix some cases where skinned mesh import get the bindpose messed up

### PR DESCRIPTION
**Pull request by judubu**
## Purpose of this PR

Ticket/Jira #: PR https://github.com/Unity-Technologies/usd-unity-sdk/pull/308 https://jira.unity3d.com/browse/USDU-269

A user pointed out that JointsMatch() functionality regressed in PR https://github.com/Unity-Technologies/usd-unity-sdk/pull/261, resulting in it returning true in nearly all cases.

Meshes can define a subset of joints or custom joints order and this fell through the cracks of the v3 refactoring/reformatting. This PR brings back the old code and add unit tests

## Testing

**Functional Testing status:**
3 new play tests checking the bindpose for: unchanged joint order, custom joint order, subset of joints

**Performance Testing status:**
Really minor performance hit as we now iterate over the joints list for for each skinned mesh.  

## Overall Product Risks

**Complexity:**

Low
**Halo Effect:**

Low
## Additional information

**Note to reviewers:**

**Reminder:**

- [ ] Add entry in CHANGELOG.md _(if applicable)_

### CODE REVIEW

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Skel/SkeletonImporter.cs [hunk 1]
```diff
@@ -203,9 +218,11 @@ static bool JointsMatch(string[] lhs, string[] rhs)
             // TODO: bind transform attribute can be animated. It's not handled yet.
             Matrix4x4 geomXf = UnityTypeConverter.FromMatrix(skinningQuery.GetGeomBindTransform());
 
-            // If the joints list is a different length than the bind transforms, then this is likely
```
Comment (vickycl) [inline package/com.unity.formats.usd/Runtime/Scripts/IO/Skel/SkeletonImporter.cs:206]: Was this change on purpose? Should it be 'is a different length than' or 'is different to'?

File: package/com.unity.formats.usd/Tests/Runtime/ImportMeshTests.cs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            var usdPath = Path.GetFullPath(AssetDatabase.GUIDToAssetPath(k_testAssetGUID));
+            var stage = UsdStage.Open(usdPath, UsdStage.InitialLoadSet.LoadNone);
+            var scene = Scene.Open(stage);
+            m_usdRoot = ImportHelpers.ImportSceneAsGameObject(scene);
+            scene.Close();
+        }
+
+        public void MatrixCompare(Matrix4x4 actual, Matrix4x4 expected, float epsilon = 1e-5f)
+        {
+            for (int i = 0; i < 16; ++i)
```
Comment (vickycl) [inline package/com.unity.formats.usd/Tests/Runtime/ImportMeshTests.cs]: I don't think this loop is necessary

**Review by vickycl — CHANGES_REQUESTED** [review]
Thanks for the fix! Added a small query about a comment and a change request in the test.

File: package/com.unity.formats.usd/Tests/Runtime/ImportMeshTests.cs (hunk 2 above)
Comment (judubu) [reply package/com.unity.formats.usd/Tests/Runtime/ImportMeshTests.cs]: Right on, stupid copy paste. Fixed

File: package/com.unity.formats.usd/Runtime/Scripts/IO/Skel/SkeletonImporter.cs (hunk 1 above)
Comment (judubu) [reply package/com.unity.formats.usd/Runtime/Scripts/IO/Skel/SkeletonImporter.cs:206]: Fixed.

File: package/com.unity.formats.usd/Tests/Editor/ImportMeshTests.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+{
+    public class ImportSkinnedMeshTests : BaseFixtureEditor
+    {
+        static string k_testAssetGUID = "3d00d71254d14bdda401019eb84373ce";
+        GameObject m_usdRoot;
+
+        [SetUp]
+        public void SetUp()
+        {
+            InitUsd.Initialize();
```
Comment (lee-aandrew) [inline package/com.unity.formats.usd/Tests/Editor/ImportMeshTests.cs]: I think since BaseFixture has a setup step that includes InitUsds.Initialize(), this one could be redundant

**Review by vickycl — APPROVED** [review]
LGTM :)

File: package/com.unity.formats.usd/Tests/Editor/ImportMeshTests.cs (hunk 3 above)
Comment (judubu) [reply package/com.unity.formats.usd/Tests/Editor/ImportMeshTests.cs]: Good point. I'll remove it.
